internal/controller/pipelinerollout_progressive.go

```diff
@@ -90,20 +92,25 @@ func (r *PipelineRolloutReconciler) processUpgradingPipelineStatus(
 		}
 	}
 
-	pipelineStatus, err := kubernetes.ParseStatus(existingUpgradingPipelineDef)
+	pipelineStatus, err := parsePipelineStatus(existingUpgradingPipelineDef)
 	if err != nil {
 		return false, fmt.Errorf("failed to parse Pipeline Status from pipeline CR: %+v, %v", existingUpgradingPipelineDef, err)
 	}
 
-	pipelinePhase := numaflowv1.PipelinePhase(pipelineStatus.Phase)
+	pipelinePhase := pipelineStatus.Phase
 	if pipelinePhase == numaflowv1.PipelinePhaseFailed {
-		err = r.updatePipelineLabel(ctx, r.restConfig, existingUpgradingPipelineDef, "")
+		// Mark the failed new pipeline recyclable.
+		err = r.updatePipelineLabel(ctx, r.restConfig, existingUpgradingPipelineDef, string(common.LabelValueUpgradeRecyclable))
```

Comment:
I am seeing the value of not deleting the Failed one, so the user can observe what happened after the fact (check Pipeline logs, etc). 

I am okay to merge, but maybe we change that in the future or make it configurable? what are your thoughts?

Reply:
oh, I see. Is it simply marked "recyclable", but then nothing causes it to pause and drain so it never actually gets deleted?